### PR TITLE
Remove taskchampion-sync-server

### PR DESCRIPTION
**Pull request by djmitche**
As mentioned in #3209, This crate has been moved to
https://github.com/GothenburgBitFactory/taskchampion-sync-server.

The integration-tests repo used the sync server to test integration between taskchampion and the sync-server. We should do that again, but after taskchampion moves to its own repo (#3209). In the interim, the cross-sync integration test can simply test syncing between local servers, but the snapshot test is no longer useful as the local server does not support snapshots.

### CODE REVIEW

File: doc/man/task-sync.5.in [hunk 1]
```diff
@@ -144,6 +144,9 @@ Users are identified by a client ID, and users with different client IDs are
 entirely independent.  Task data is encrypted by Taskwarrior, and the sync
 server never sees un-encrypted data.
 
+The server is developed in
+https://github.com/GothenburgBitFactory/taskchampion-sync-server.
+
```
Comment (ryneeverett) [inline doc/man/task-sync.5.in:149]: Would it make sense to remove the rest of the section -- from this point to the end? It seems a bit odd to document the command line interface here now that it's a separate project.

File: taskchampion/CONTRIBUTING.md [hunk 2]
```diff
@@ -39,13 +36,13 @@ Aside from that, start reading the docs and the source to learn more!
...
 It is linked from the README.
 
-There are three crates in this repository.
+There are three important crates in this repository.
 You may be able to limit the scope of what you need to understand to just one crate.
  * `taskchampion` is the core functionality of the application, implemented as a library
- * `taskchampion-cli` implements the command-line interface (in `cli/`)
- * `taskchampion-sync-server` implements the synchronization server (in `sync-server/`)
+ * `taskchampion-lib` implements a C API for `taskchampion`, used by Taskwarrior
+ * `integration-tests` contains some tests for integrations between multiple crtes.
```
Comment (ryneeverett) [inline taskchampion/CONTRIBUTING.md]:
```suggestion
 * `integration-tests` contains some tests for integrations between multiple crates.
```

RFC: Install Codespell in @djmitche's Editor :)

File: taskchampion/CONTRIBUTING.md [hunk 3]
```diff
@@ -24,11 +24,8 @@ Other ideas;
 
 TaskChampion is a typical Rust application.
 To work on TaskChampion, you'll need to [install the latest version of Rust](https://www.rust-lang.org/tools/install).
-Once you've done that, run `cargo build` at the top level of this repository to build the binaries.
```
Comment (ryneeverett) [inline taskchampion/CONTRIBUTING.md:27]: Would it make sense to say here that the build is now typically invoked through cmake?

File: taskchampion/integration-tests/tests/snapshots.rs [hunk 4]
```diff
@@ -1,98 +0,0 @@
...
-
-        assert_eq!(t11.get_description(), "sync-back");
-
-        Ok(())
-    }
-
-    let port = server().await?;
-    actix_rt::task::spawn_blocking(move || client(port)).await??;
-    Ok(())
-}
```
Comment (ryneeverett) [inline taskchampion/integration-tests/tests/snapshots.rs:98]: Would it be worth it to make taskchampion-sync-server an external dependency of the integration tests?

**Review by ryneeverett — APPROVED** [review]

File: doc/man/task-sync.5.in (hunk 1 above)
Comment (djmitche) [reply doc/man/task-sync.5.in:149]: That makes sense.

File: taskchampion/CONTRIBUTING.md (hunk 2 above)
Comment (djmitche) [reply taskchampion/CONTRIBUTING.md]: I really should!

File: taskchampion/CONTRIBUTING.md (hunk 3 above)
Comment (djmitche) [reply taskchampion/CONTRIBUTING.md:27]: Once this is a separate project, that won't make sense. This PR is just trying to remove sync-server, with more to come in #3209.

File: taskchampion/integration-tests/tests/snapshots.rs (hunk 4 above)
Comment (djmitche) [reply taskchampion/integration-tests/tests/snapshots.rs:98]: Once Taskchampion is moved to its own repo, and the sync-server is released, I think that will make sense. The other option is to submodule the integration tests in both the taskchampion and taskchampion-sync-server repos.

**Pull request merged**